algorithms/unit_tests/TestStdAlgorithmsTeamTransformUnaryOp.cpp

```diff
@@ -0,0 +1,182 @@
+  // -----------------------------------------------
+  using space_t = Kokkos::DefaultExecutionSpace;
+  Kokkos::TeamPolicy<space_t> policy(numTeams, Kokkos::AUTO());
+  // create the destination view
+  Kokkos::View<ValueType**> destView("destView", numTeams, numCols);
+  // make a host copy of the dest view that we can check below
+  // to be all zeros since this should remain unchanged
+  auto destViewBeforeOp_h = create_host_space_copy(destView);
+
+  // each team stores the distance of the returned iterator from the
+  // beginning of the interval that team operates on and then we check
+  // that these distances match the expectation
+  Kokkos::View<std::size_t*> distancesView("distancesView", numTeams);
+  // sentinel to check if all members of the team compute the same result
+  Kokkos::View<bool*> intraTeamSentinelView("intraTeamSameResult", numTeams);
+
+  // use CTAD for functor
+  TestFunctorA fnc(sourceView, destView, distancesView, intraTeamSentinelView,
+                   apiId);
+  Kokkos::parallel_for(policy, fnc);
+
+  // -----------------------------------------------
+  // check
+  // -----------------------------------------------
+  auto distancesView_h         = create_host_space_copy(distancesView);
+  auto sourceViewAfterOp_h     = create_host_space_copy(sourceView);
+  auto destViewAfterOp_h       = create_host_space_copy(destView);
+  auto intraTeamSentinelView_h = create_host_space_copy(intraTeamSentinelView);
+  for (std::size_t i = 0; i < destViewBeforeOp_h.extent(0); ++i) {
+    for (std::size_t j = 0; j < destViewBeforeOp_h.extent(1); ++j) {
+      // source view should not change
+      ASSERT_EQ(sourceViewAfterOp_h(i, j), cloneOfSourceViewBeforeOp_h(i, j));
+
+      // elements in dest view should be the source elements plus two
+      ASSERT_EQ(destViewAfterOp_h(i, j), cloneOfSourceViewBeforeOp_h(i, j) + 2);
+      ASSERT_EQ(destViewBeforeOp_h(i, j), ValueType(0));
+    }
+
+    // each team should return an iterator whose distance from the
+    // beginning of the row equals the num of columns since
+    // each team transforms all elements in each row
+    EXPECT_TRUE(distancesView_h(i) == numCols);
+    ASSERT_TRUE(intraTeamSentinelView_h(i));
+  }
+}
+
+template <class LayoutTag, class ValueType>
+void run_all_scenarios() {
+  for (int numTeams : teamSizesToTest) {
+    for (const auto& numCols : {0, 1, 2, 13, 101, 1444, 11113}) {
+      for (int apiId : {0, 1}) {
+        test_A<LayoutTag, ValueType>(numTeams, numCols, apiId);
+      }
+    }
+  }
+}
+
+TEST(std_algorithms_transform_team_test, test_unary_op) {
+  run_all_scenarios<DynamicTag, double>();
+  run_all_scenarios<StridedTwoRowsTag, int>();
+  run_all_scenarios<StridedThreeRowsTag, unsigned>();
+}
+
+// TEST(std_algorithms_transform_team_test, test_binary_op) {
+//   run_all_scenarios<DynamicTag, double>(1);
+//   run_all_scenarios<StridedTwoRowsTag, int>(1);
+//   run_all_scenarios<StridedThreeRowsTag, unsigned>(1);
+// }
```

Comment:
Why is this commented?

Reply:
my bad for not removing it, i just did. Originally the test was called `TestStdAlgorithmsTeamTransform.cpp` and contained both the test for the unary and binary op. Then i decided to split it so now there are two files , one testing the unary and one testing the binary op. I forgot to remove this code above thanks for pointing it out